Assets/SEE/Controls/Actions/ContextMenuAction.cs

```diff
@@ -160,6 +168,24 @@ private static TreeWindow ActivateTreeWindow(GraphElement graphElement, Transfor
             return openWindow;
         }
 
+        /// <summary>
+        /// Returns a MetricWindow showing the attributes of the given graph element.
+        /// </summary>
+        /// <param name="graphElementRef">The graph element to activate the metric window for</param>
+        /// <returns></returns>
+        private static MetricWindow CreateMetricWindow(GraphElementRef graphElementRef)
+        {
+            // Create new window for active selection, or use existing one
+            if (!graphElementRef.TryGetComponent(out MetricWindow metricMenu))
+            {
+                metricMenu = graphElementRef.gameObject.AddComponent<MetricWindow>();
+
+                metricMenu.Title = "MetricWindow " + graphElementRef.Elem.ToShortString();
+                metricMenu.graphElement = graphElementRef.Elem;
+            }
+            return metricMenu;
```

Comment:
Do you mean returned instead of removed? If yes then null can't be returned here because it either uses an existing one from "if (!graphElementRef.TryGetComponent(out MetricWindow metricMenu))" or a new component will be added and initialized and metricMenu will refer to this newly created instance.